indexer/src/test/java/au/org/aodn/esindexer/service/VocabServiceTest.java

```diff
@@ -0,0 +1,271 @@
+
+import java.io.IOException;
+import java.util.*;
+
+// JSONAssert is a useful dependency for comparing JSON values, replacing the traditional string-to-string approach when dealing with JSON.
+// More details: https://www.baeldung.com/jsonassert#overview, https://github.com/skyscreamer/JSONassert
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest(
+        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+@ActiveProfiles("test")
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class VocabServiceTest extends BaseTestClass {
+
+    @Autowired
+    VocabService vocabService;
+
+    @Autowired
+    protected ObjectMapper indexerObjectMapper;
+
+    @Test
+    void testExtractParameterVocabLabelsFromThemes() throws IOException {
+        // Prepare themes
+        List<ThemesModel> themes = List.of(
+                new ThemesModel(Arrays.asList(
+                        new ConceptModel("Temperature of the water body", "http://vocab.nerc.ac.uk/collection/P01/current/TEMPPR01"),
+                        new ConceptModel("Practical salinity of the water body", "http://vocab.nerc.ac.uk/collection/P01/current/PSLTZZ01"),
+                        new ConceptModel("Concentration of carbon (total inorganic) per unit mass of the water body", "http://vocab.aodn.org.au/def/discovery_parameter/entity/1"),
+                        new ConceptModel("Total alkalinity per unit mass of the water body", "http://vocab.nerc.ac.uk/collection/P01/current/MDMAP014"),
+                        new ConceptModel("Saturation state of aragonite in the water body", "http://vocab.aodn.org.au/def/discovery_parameter/entity/24"),
+                        new ConceptModel("Saturation state of aragonite in the water body", "http://vocab.aodn.org.au/def/discovery_parameter/entity/24"),
+                        new ConceptModel("pH (total scale) of the water body", "http://vocab.aodn.org.au/def/discovery_parameter/entity/27")
+                ), "theme", null, "AODN Discovery Parameter Vocabulary")
+        );
+
+        // Perform the test
+        List<String> parameterVocabs = vocabService.extractVocabLabelsFromThemes(themes, AppConstants.AODN_DISCOVERY_PARAMETER_VOCABS);
+
+        // Assertions
+        assertNotNull(parameterVocabs);
+        assertTrue(parameterVocabs.stream().anyMatch(vocab -> vocab.equalsIgnoreCase("alkalinity")));
+        assertTrue(parameterVocabs.stream().anyMatch(vocab -> vocab.equalsIgnoreCase("temperature")));
+        assertTrue(parameterVocabs.stream().anyMatch(vocab -> vocab.equalsIgnoreCase("salinity")));
+        assertTrue(parameterVocabs.stream().anyMatch(vocab -> vocab.equalsIgnoreCase("carbon")));
+        assertTrue(parameterVocabs.stream().anyMatch(vocab -> vocab.equalsIgnoreCase("pH (total scale) of the water body")));
```

Comment:
so this might affect the UI design to display phrase-like selecting options

the second-level vocabs are not always short words, the current implementation to process ardc vocabs has bug overlooking vocabs that don't have leaf nodes (bottom-level).

I manually inspected the ardc apis to verify this, the current prod portal also has "pH (total scale) of the water body"

![image](https://github.com/user-attachments/assets/ba1ef321-3af9-4a4d-9379-b7e366685359)

Reply:
Well I am a bit lost on what you mean, is this a requirement ?

Reply:
no I meant what we currently have for parameter vocab has bug, this current PR you're reviewing will fix that.

the bug missed identifying some second-level vocabs

Reply:
and because some of them are very long, not just 1 or 2 words, when it appears in the new portal 

in particular, here:
![image](https://github.com/user-attachments/assets/6a1587cb-f01d-4a2a-a655-508c4d0a5681)

it may look ugly?

Reply:
e.g `pH (total scale) of the water body` has never been identified as second-level vocab (which should be) until this PR.